walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,150 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Iterator; 
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> requestAttendees = request.getAttendees();
+    Collection<String> optionalAttendees = request.getOptionalAttendees();
+    Collection<String> allAttendees = new ArrayList<>();
+    allAttendees.addAll(requestAttendees);
+    allAttendees.addAll(optionalAttendees);
+    long requestDuration = request.getDuration();
+    int numRequestSlots = (int) requestDuration / 30;
+    int numEvents = events.size();
+
+    boolean[] mandatoryTimeSlots = new boolean[48];
+    boolean[] allTimeSlots = new boolean[48];
```

Comment:
You define these here, but below re-assign them before using them, so you can probably delete this.